src/html-media-source.js

```diff
@@ -231,10 +231,22 @@ export default class HtmlMediaSource extends videojs.EventTarget {
     let buffer;
     let parsedType = parseContentType(type);
 
-    // create a virtual source buffer to transmux MPEG-2 transport
+    // Create a VirtualSourceBuffer to transmux MPEG-2 transport
     // stream segments into fragmented MP4s
     if (parsedType.type === 'video/mp2t') {
-      // default codecs
+      if (this.sourceBuffers.length !== 0) {
+        // If another VirtualSourceBuffer already exists, then we are creating a
+        // SourceBuffer for an alternate audio track and therefore we know that
+        // the source has both an audio and video track.
+        // That means we should trigger the manual creation of the real
+        // SourceBuffers instead of waiting for the transmuxer to return data
```

Comment:
The built in assumption is that the first VSB is muxed unless we are told otherwise. If we create a second source buffer, it is fairly safe to assumed that it is audio-only. This assumption is built into HLS and MS right now.

Internally, we have a way transfer (or share) ownership of native SourceBuffers that are for audio data but we do not have a way to transfer (or share) ownership of native video SourceBuffers.